src/libcore/macros.rs

```diff
@@ -56,7 +64,8 @@ macro_rules! panic {
 macro_rules! assert {
     ($cond:expr) => (
         if !$cond {
-            panic!(concat!("assertion failed: ", stringify!($cond)))
```

Comment:
It is necessary... it is avoiding having a string literal with embedded curly braces (that results from macro-expanding the `concat!` with `stringify!`) being fed into the `panic!` directly ... or are you claiming that is somehow already being handled?